src/optimum/nvidia/models/llama.py

```diff
@@ -157,8 +140,8 @@ def convert(
     def allocate_model(
         config: ModelConfig, sharding: ShardingConfig, dtype: DataType, quant_mode: QuantMode
     ) -> Module:
-        LOGGER.debug(f"Allocating {LLaMAForCausalLM.__name__} model")
-        return LLaMAForCausalLM(
+        LOGGER.debug(f"Allocating {LlamaWeightAdapter.TENSORRT_LLM_MODEL_CLASS.__name__} model")
+        return LlamaWeightAdapter.TENSORRT_LLM_MODEL_CLASS(
```

Comment:
Here we use `TENSORRT_LLM_MODEL_CLASS` as this class attribute is also used in the base class - align everywhere on it.